### PR TITLE
neutrino: misc connection management fixes

### PR DESCRIPTION
**Pull request by wpaulino**
In this PR, we include a series of connection management fixes that were discovered through testing chain syncs with no `connect` or `addpeers`. Each fix has been included in its own commit along with a descriptive message of the issue it's addressing.

Mainly, this PR attempts to ensure we can more aggressively connect to peers through the underlying connection manager.

### CODE REVIEW

File: blockmanager.go [hunk 1]
```diff
@@ -1273,9 +1273,13 @@ func (b *blockManager) resolveConflict(
 					log.Errorf("Unable to ban peer %v: %v",
 						peer, err)
 				}
+				if sp := b.server.PeerByAddr(peer); sp != nil {
```
Comment (Roasbeef) [inline blockmanager.go]: Could've sworn that `BanPeer` _did_ disconnect the peer, but this isn't the case as is...

File: neutrino.go [hunk 2]
```diff
@@ -1360,15 +1356,16 @@ func (s *ChainService) outboundPeerConnected(c *connmgr.ConnReq, conn net.Conn)
...
-		conn.Close()
+		s.connManager.Remove(c.ID())
 		return
 	}
 
 	sp := newServerPeer(s, c.Permanent)
 	p, err := peer.NewOutboundPeer(newPeerConfig(sp), peerAddr)
 	if err != nil {
 		log.Debugf("Cannot create outbound peer %s: %s", c.Addr, err)
-		s.connManager.Disconnect(c.ID())
```
Comment (Roasbeef) [inline neutrino.go]: dat connection leak...

File: neutrino.go [hunk 3]
```diff
@@ -1360,15 +1356,16 @@ func (s *ChainService) outboundPeerConnected(c *connmgr.ConnReq, conn net.Conn)
 	// If we're already connected to this peer, then we'll close out the new
 	// connection and keep the old.
 	if s.PeerByAddr(peerAddr) != nil {
-		conn.Close()
+		s.connManager.Remove(c.ID())
```
Comment (Roasbeef) [inline neutrino.go]: We may want to eventually change the names of `Remove/Disconnect` in the `connmgr` to better reflect their retry semantics.

File: blockmanager.go (hunk 1 above)
Comment (halseth) [reply blockmanager.go]: > Could modify the method to do so as well

+1

Comment (wpaulino) [reply blockmanager.go]: Done.

File: log.go [hunk 4]
```diff
@@ -37,6 +38,7 @@ func UseLogger(logger btclog.Logger) {
 	addrmgr.UseLogger(logger)
 	blockntfns.UseLogger(logger)
 	pushtx.UseLogger(logger)
+	connmgr.UseLogger(logger)
```
Comment (Roasbeef) [inline log.go]: FWIW, when it's run within `lnd`, we've already hooked up the `connmgr` as a logger.

File: blockmanager.go [hunk 5]
```diff
@@ -1363,10 +1350,6 @@ func (b *blockManager) resolveConflict(
 					log.Errorf("Unable to ban peer %v: %v",
 						peer, err)
 				}
-				sp := b.server.PeerByAddr(peer)
-				if sp != nil {
```
Comment (Roasbeef) [inline blockmanager.go]: 👍 

much less clutter this way by pushing the logic into `BanPeer`.

**Pull request merged**